### PR TITLE
[GlobalISel] Lowering of {get,set,reset}_fpenv

### PR DESCRIPTION
**Pull request by spavloff**
The intrinsics get_fpenv, set_fpenv and reset_fpenv in this change are implemented as calls to math library functions. Target specific lowering will be implemented later on.

### CODE REVIEW

File: llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp [hunk 1]
```diff
@@ -1122,6 +1122,8 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
   getActionDefinitionsBuilder(G_FMAD).lower();
 
   // Access to floating-point environment.
+  getActionDefinitionsBuilder({G_GET_FPENV, G_SET_FPENV, G_RESET_FPENV})
```
Comment (tschuett) [inline llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp:1125]: You could merge them into one builder. They have similar concepts and similar legalization strategies.

File: llvm/test/CodeGen/AArch64/fpenv.ll [hunk 2]
```diff
@@ -62,4 +62,47 @@ define void @func_set_rounding_downward() {
   ret void
 }
 
+define i64 @get_fpenv_01() nounwind {
```
Comment (tschuett) [inline llvm/test/CodeGen/AArch64/fpenv.ll:65]: This is presumably not testing what you planned. The RUN line does not contain the string `globalisel`. It is testing DAG code generation. If you look at the sister file xtn.ll, it has two RUN lines to test code generation for both selectors and highlight their differences.

**Review by aemerson — APPROVED** [review]
Looks ok, if there's an existing codegen test in AArch64 for these intrinsic could you add RUN lines for GISel?

File: llvm/test/CodeGen/AArch64/GlobalISel/legalize-fpenv.mir [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+# RUN: llc -mtriple=aarch64-linux-gnu -run-pass=legalizer %s -o - | FileCheck %s
+
+--- |
+  target triple = "aarch64-unknown-linux-gnu"
+
+  declare i64 @llvm.get.fpenv.i64()
+  declare void @llvm.set.fpenv.i64(i64)
+  declare void @llvm.reset.fpenv()
+
+  define i64 @func_get_fpenv() {
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/GlobalISel/legalize-fpenv.mir:11]: Don't need the IR section

File: llvm/test/CodeGen/AArch64/GlobalISel/legalize-fpenv.mir [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    call void @llvm.reset.fpenv()
+    ret void
+  }
+
+...
+---
+name:            func_get_fpenv
+tracksRegLiveness: true
+body:             |
+  bb.1.entry:
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/GlobalISel/legalize-fpenv.mir:34]: bb.0, drop .1.entry

**Review by arsenm — APPROVED** [review]

**Pull request merged**